frontend-react/src/hooks/UsePaginator.tsx

```diff
@@ -0,0 +1,101 @@
+/* Pagination behavior:
```

Comment:
LOVE the detailed comments.

Suggestion only: Pictures might be helpful? Maybe create a README.md or add a section somewhere that discusses the approach in a more rich format other than plaintext then link to it from code?

Reply:
This is another sandbox item on my list; I wanna get it out of our project and make it as pure and reusable as I can, and part of that free-time work should result in both a suite of tests and a nice README we can pull into the project, similar to the `useEndpoint` stuff I did this past weekend.